build/build/src/engine/context.rs

```diff
@@ -345,11 +345,16 @@ impl RunContext {
 
         if !tasks.is_empty() {
             debug!("Building distributions and native images.");
+            let mut sbt_arg;
             if crate::ci::big_memory_machine() {
-                sbt.call_arg(Sbt::concurrent_tasks(tasks)).await?;
+                sbt_arg = Sbt::concurrent_tasks(tasks);
             } else {
-                sbt.call_arg(Sbt::sequential_tasks(tasks)).await?;
+                sbt_arg = Sbt::sequential_tasks(tasks);
             }
+            if self.config.clean_before {
+                sbt_arg = Sbt::sequential_tasks(vec!["clean", &sbt_arg]);
```

Comment:
Shouldn't the clean only run for those who modify `engine` sources? It should be possible to detect what files are changed in the PR: https://stackoverflow.com/questions/74265821/get-modified-files-in-github-actions

Reply:
It does not matter if the current branch modified engine sources or not. It can be the case that the runner that picked up the job previously was building the branch that did and has an SBT cache left from building that branch.